src/requestQueue.js

```diff
@@ -17,6 +17,12 @@ export default class RequestQueue {
 
```

Comment:
Could we just simplify this class similar to my implementation? Seems like this is way more complicated that needed. It is up to you though